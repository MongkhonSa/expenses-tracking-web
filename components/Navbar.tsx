import { Button, Col, Row } from "antd";
import Link from "next/link";
import { useRouter } from "next/router";
import React from "react";
import styled from "styled-components";

const LogoutComponent = styled(Col)``;

function Navbar() {
  const router = useRouter();
  return (
    <Row justify="end">
      <LogoutComponent>
        <Button>
          <Link href="/">Home</Link>
        </Button>
      </LogoutComponent>
      <LogoutComponent>
        <Button>
          <Link href="/report">Report</Link>
        </Button>
      </LogoutComponent>
      <LogoutComponent>
        <Button
          onClick={() => {
            localStorage.clear();
            router.push("/login");
          }}
        >
          <h4>logout</h4>
        </Button>
      </LogoutComponent>
    </Row>
  );
}

export default Navbar;
