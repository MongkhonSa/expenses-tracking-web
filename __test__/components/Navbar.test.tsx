import { fireEvent, render, screen } from "@testing-library/react";
import Navbar from "../../components/Navbar";
import mockRouter from "../../mock/routerMock";

jest.mock("next/router", () => ({
  useRouter() {
    return mockRouter;
  },
}));
jest.spyOn(require("next/router"), "useRouter");

describe("Navbar", () => {
  beforeEach(() => {});
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders Navbar unchanged", () => {
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });
  it("should reset local storage when logout", () => {
    render(<Navbar />);
    fireEvent.click(screen.getByText("logout"));
    expect(localStorage.clear).toBeCalledTimes(1);
    expect(mockRouter.push).toHaveBeenCalledWith("/login");
  });
});
