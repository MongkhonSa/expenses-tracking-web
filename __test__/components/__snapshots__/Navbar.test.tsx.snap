// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders Navbar unchanged 1`] = `
<div>
  <div
    class="ant-row ant-row-end"
  >
    <div
      class="ant-col sc-724045b4-0"
    >
      <button
        class="ant-btn ant-btn-default"
        type="button"
      >
        <a
          href="/"
        >
          Home
        </a>
      </button>
    </div>
    <div
      class="ant-col sc-724045b4-0"
    >
      <button
        class="ant-btn ant-btn-default"
        type="button"
      >
        <a
          href="/report"
        >
          Report
        </a>
      </button>
    </div>
    <div
      class="ant-col sc-724045b4-0"
    >
      <button
        class="ant-btn ant-btn-default"
        type="button"
      >
        <h4>
          logout
        </h4>
      </button>
    </div>
  </div>
</div>
`;
